src/main/kotlin/org/jetbrains/research/ml/ast/transformations/commands/DeleteCommand.kt

```diff
@@ -0,0 +1,239 @@
+package org.jetbrains.research.ml.ast.transformations.commands
+
+import com.intellij.application.options.CodeStyle
+import com.intellij.formatting.FormattingContext
+import com.intellij.formatting.FormattingModel
+import com.intellij.lang.LanguageFormatting
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiParserFacade
+import com.intellij.psi.PsiWhiteSpace
+import com.intellij.psi.codeStyle.CodeStyleSettings
+import com.jetbrains.python.psi.LanguageLevel
+import com.jetbrains.python.psi.PyElementGenerator
+import java.util.concurrent.Callable
+
+
+
+
+
+
+/**
+ * Represents a range of adjacent siblings from [startPsi] (inclusive) to [endPsi] (inclusive),
+ * siblings are the first (prev, next) siblings that meet a [condition]
+ */
+class RestorablePsiElement(private var psiElement: PsiElement) {
+
+    enum class Type {
+        ONLY_CHILD, LEFT_CHILD, RIGHT_CHILD, MIDDLE_CHILD
+    }
+
+    private var psiText: String = psiElement.text
+    private val project = psiElement.project
+
+    private var prevSibling: PsiElement? = findNextSiblingOnCondition(
+        { it.prevSibling },
+        { this.prevSibling = it.newPsi }
+    )
+    private var nextSibling: PsiElement? = findNextSiblingOnCondition(
+        { it.nextSibling },
+        { this.nextSibling = it.newPsi }
+    )
+    private var parent = findParent()
+    private val type = findType()
```

Comment:
If you want to delete an element and suddenly forget about its existence it's strange :) 
I'd rather not make them extensions because there will be problems with implementation: for example. `this.parent` in implementation of findParent() will be ambiguous, since this will refer to the PsiElement, and not to the PsiRestorableElement

I also cannot find the type of any PsiElement, only of psiElement, since I need to know its siblings. So it would be strange to have an extension